deepspeed/runtime/engine.py

```diff
@@ -1076,7 +1077,7 @@ def _configure_fp16_optimizer(self, optimizer):
     def _configure_zero_optimizer(self, optimizer):
         zero_stage = self.zero_optimization_stage()
         log_dist('Creating fp16 ZeRO stage {} optimizer'.format(zero_stage), ranks=[0])
-        assert not self.allreduce_always_fp32(), "ZeRO does not support 'fp32_allreduce': true"
+        assert self.communication_data_type is None, "ZeRO supports only 'communication_data_type': 'none'"
```

Comment:
The assert is specifically to reflect that fp32 all-reduce is not supported in ZeRO and is independent of considerations about valid configuration values for `communication_data_type`. In future, when ZeRO starts to support fp32 all-reduce it will much easier to remove this assert without worrying of other side effects. Does that make sense?